src/libraries/System.Diagnostics.DiagnosticSource/tests/PropagatorTests.cs

```diff
@@ -527,6 +527,14 @@ private Activity CreateW3CActivity(string name, string state, IEnumerable<KeyVal
             return list;
         }
 
+        [Fact]
+        public void TestBuiltInPropagatorsAreCached()
+        {
+            Assert.Same(TextMapPropagator.CreateDefaultPropagator(), TextMapPropagator.CreateDefaultPropagator());
```

Comment:
>Assert [](http://example.com/codeflow?start=12&length=6)

do we really need to check for that? that will put restriction in the future to have us create a new instances if needed.

Reply:
Would we ever not return the same instance for parameterless factories?

The implementation here makes use of this assertion to make some minor optimizations, but would continue to function correctly even if that were not the case.

Reply:
> Would we ever not return the same instance for parameterless factories?

No-one knows what can happen in the future. imagine in the future we allow adding setter properties to the propagators which will require us always create a new instance at that time.

Reply:
Fair point, but we can always change the asserts if that happens. The test is looking at an implementation detail, we are not promising the behavior by documenting it.

Reply:
I am not worried much about the test more than the product code taking dependencies on that. I am trying to avoid in the future if something change in the propagator side will require to change http client.

Reply:
Removed any reliance on reference equality from the implementation. The test is fine to stay as-is.